File: hejoo345/week_17/PGS_정수 삼각형.js
function solution(triangle) {
    let dp = [triangle[0][0]];

    for (let i = 1; i < triangle.length; i++) {
        let tmp = [];
        for (let j = 0; j < i + 1; j++) {
            const left = Number(triangle[i][j]) + Number(dp[j - 1]);
            const right = Number(triangle[i][j]) + Number(dp[j]);
            if (j == 0) {
                tmp.push(right);
            } else if (j == i) {
                tmp.push(left);
            } else {
                let max = Math.max(left, right);
                tmp.push(max);
            }
        }
        dp = tmp;
    }
    return Math.max(...dp);
}

File: raeunlee/week_17/PGS_정수삼각형.py
def solution(triangle):
    
    for i in range(1, len(triangle)): # 둘째 줄부터 끝까지 
        for j in range(i+1):
            
            if j == 0: # 첫번째 수
                triangle[i][j] += triangle[i-1][j]
                
            elif j == i: # 마지막 수
                triangle[i][j] += triangle[i-1][j-1]
                
            else: # 그 전에꺼 대각선 방향중 더 큰것 더하기
                triangle[i][j] += max(triangle[i-1][j-1], triangle[i - 1][j])
                
    return max(triangle[len(triangle)-1])